### PR TITLE
1.14 Aliases Update

### PR DESCRIPTION
**Pull request by APickledWalrus**
This is a rather large PR, I expect it to possibly have a few issues. This PR was done using snapshot 19w14a.
> Added all 1.14 blocks and items
> Added support for the "shape" block state for stairs

### CODE REVIEW

**Comment by abandonedaccount6235** [comment]
nice

**Comment by ShaneBeee** [comment]
Wow, very nice job.
Ps Just skimming thru, the only one I can find that you missed was the jigsaw block

File: decoration.sk [hunk 1]
```diff
@@ -16,6 +16,76 @@ unchanged decoratives:
...
+	#Whether the grindstone is on the floor, wall, or ceiling.
+	{grindstone face}:
+		{default} = -
+		floor = -[face=floor]
+		wall = -[face=wall]
+		ceiling - [face=ceiling]
+	{directional} {grindstone face} grindstone¦s = minecraft:grindstone
+
+	#Non-functional "crafting" blocks
+	cartography table¦s = minecraft:cartography_table
```
Comment (ShaneBeee) [inline decoration.sk]: Not that this makes any difference, but the cartography table **does** have a function

Comment (APickledWalrus) [reply decoration.sk]: Oops! Must have accidentally moved that one when I was sorting through them.

File: combat.sk [hunk 2]
```diff
@@ -96,8 +96,9 @@ horse armor:
 	iron (horse armo[u]r|barding) = minecraft:iron_horse_armor
 	gold[en] (horse armo[u]r|barding) = minecraft:golden_horse_armor
 	diamond (horse armo[u]r|barding) = minecraft:diamond_horse_armor
+	leather (horse armo[u]r|barding) = minecraft:leather_horse_armor
```
Comment (TheBentoBox) [inline combat.sk:104]: This has to be in a 1.14-specific category

File: combat.sk [hunk 3]
```diff
@@ -96,8 +96,9 @@ horse armor:
...
 
-	[any] (horse armo[u]r|barding) = iron horse armor, gold horse armor, diamond horse armor
+	[any] (horse armo[u]r|barding) = iron horse armor, gold horse armor, diamond horse armor, leather horse armor
```
Comment (TheBentoBox) [inline combat.sk:106]: And same for this, we should have a separate 1.14+ alias group that includes the leather horse armor

File: decoration.sk [hunk 4]
```diff
@@ -16,6 +16,76 @@ unchanged decoratives:
...
+
+	{scaffold bottom}:
+		{default} = -
+		bottomless = -[bottom=false]
+		bottom = -[bottom=true]
+	{waterloggable} {scaffold bottom} scaffolding = minecraft:scaffolding
+
+	#Crafting Blocks
+	{directional} loom¦s = minecraft:loom
+	{barrel open}:
```
Comment (TheBentoBox) [inline decoration.sk]: There's already an `{openable}` global variation you can use in place of this instead.

File: foodstuffs.sk [hunk 5]
```diff
@@ -30,6 +30,11 @@ unchanged foods:
 	rotten flesh = minecraft:rotten_flesh
 	spider eye¦s = minecraft:spider_eye
 
+village and pillage update foods:
+	minecraft version = 1.14 or newer
+	dried kelp = minecraft:dried_kelp
+	sweet berries = minecraft:sweet_berries
```
Comment (TheBentoBox) [inline foodstuffs.sk]: I'd make this `berr(y|ies)` so people can refer to a single one singularly.

File: misc.sk [hunk 6]
```diff
@@ -10,6 +10,43 @@ mining:
...
+		(stagnant|stage 0) = -[stage=0]
+		(growing|stage 1) = -[stage=1]
+	{bamboo age} {bamboo leaves} {bamboo stage} bamboo = minecraft:bamboo
+	bamboo = minecraft:bamboo
+
+	flower charge banner pattern¦s = minecraft:flower_banner_pattern
+	creeper charge banner pattern¦s = minecraft:creeper_banner_pattern
+	skull charge banner pattern¦s = minecraft:skull_banner_pattern
+	(mojang|thing) banner pattern¦s = minecraft:mojang_banner_pattern
+	globe banner pattern¦s = minecraft:globe_banner_pattern
```
Comment (TheBentoBox) [inline misc.sk:56]: Maybe a grouped alias for banner patterns?

File: misc.sk [hunk 7]
```diff
@@ -174,7 +212,15 @@ old music discs:
 	[music] (disc|record) ward = minecraft:record_ward
 	any [music] (disc|record) = music disc 11, music disc 13, music disc blocks, music disc cat, music disc chirp, music disc far, music disc mall, music disc mellohi, music disc stal, music disc strad, music disc wait, music disc ward
 
-new dyes:
+village and pillage update dyes:
```
Comment (TheBentoBox) [inline misc.sk]: For this to work (the 1.14 one to properly override) it needs to be declared below the other dye sections

File: misc.sk [hunk 8]
```diff
@@ -215,6 +261,17 @@ old dyes:
...
+
+village and pillage spawn eggs:
+	minecraft version = 1.14 or newer
+	(pillager [spawn] egg|spawn pillager)¦s = minecraft:pillager_spawn_egg
+	(ravager [spawn] egg|spawn ravager)¦s = minecraft:ravager_spawn_egg
+	(wandering trade [spawn] egg|spawn wandering trader)¦s = minecraft:wandering_trader_spawn_egg
+	(trader llama [spawn] egg|spawn trader llama)¦s = minecraft:trader_llama_spawn_egg
+	(cat [spawn] egg|spawn cat)¦s = minecraft:cat_spawn_egg #Cats are now a seperate mob from ocelots
+	(fox [spawn] egg|spawn fox)¦s = minecraft:fox_spawn_egg
+	(panda [spawn] egg|spawn panda)¦s = minecraft:panda_spawn_egg
```
Comment (TheBentoBox) [inline misc.sk]: We'll also need an `any spawn egg` that includes the 1.14 eggs.

**Review by TheBentoBox — CHANGES_REQUESTED** [review]
Overall looks really solid, great job. I had some of this done locally already (as we discussed) but not nearly all of this and the quality looks really good. Just a few suggestions/fixes and I think it'll be ready.

**Comment by APickledWalrus** [comment]
Okay, sounds good. I will make those changes when I get the chance later today.

EDIT: Changes have been made.

File: misc.sk [hunk 9]
```diff
@@ -10,6 +10,52 @@ mining:
...
+		(some berries|berried) = -[age=2]
+		(many berries|very berried) = -[age=3]
+	{berry age} sweet berry bush [plant] = minecraft:sweet_berry_bush
+
+	flower charge banner pattern¦s = minecraft:flower_banner_pattern
+	creeper charge banner pattern¦s = minecraft:creeper_banner_pattern
+	skull charge banner pattern¦s = minecraft:skull_banner_pattern
+	(mojang|thing) banner pattern¦s = minecraft:mojang_banner_pattern
+	globe banner pattern¦s = minecraft:globe_banner_pattern
+	[any] banner pattern¦s = flower charge banner pattern, creeper charge banner pattern, skull charge banner pattern, thing banner pattern, globe banner pattern,
```
Comment (TheBentoBox) [inline misc.sk]: Is the extra comma at the end of this line harmless? I actually have no idea how the aliases handle that.

**Review by TheBentoBox — APPROVED** [review]

**Pull request merged**